### DISCUSSION TITLE
Select category OR add a new option.

### DISCUSSION THREAD

**Discussion by validlogin** [Budibase/budibase#7468] [Ideas]
Greetings!

Is there a way to have a dropdown of options available OR to manually type in a new/custom option?

For example, we have a drop-down of currently available categories. However, if we want to add a new category, there doesn't appear to be a way to do this with the dropdown.

We currently achieve this with a button that changes the dropdown to a field:
![image](https://user-images.githubusercontent.com/63905362/186833538-ff351b18-d456-446f-9e81-e52d4d2c74ae.png)
![image](https://user-images.githubusercontent.com/63905362/186833998-78f1f7a6-13fd-4b9d-a0c4-a50f7bb169bb.png)

But it would be awesome if we could have this be integrated into one element and not have a hacky solution to this? :)

**Comment by melohagan** [comment]
Hi @validlogin 

Have you had a look at the **Autocomplete** option? https://docs.budibase.com/docs/option-picker#adding-autocomplete

**Reply by validlogin** [reply]
@melohagan I did see this feature but I don’t believe it achieves what we are looking for, unless I am mistaken? 

From my understanding, the autocomplete option allows you to type in and select an already existing option; however, you cannot use this to type in custom text and have the submitted field be your custom text, correct?

For clarification, this is used when adding a new row to a table (in my case, a MySQL table). The dropdown pulls existing values from the table and presents them to the user.

Thank you!

**Reply by validlogin** [reply]
Here's an example of a previous program we would use that would allow us to do this. We can either manually type in what we would like to use in the field, or we could select the dropdown and choose an already existing category.

Thanks!

![image](https://user-images.githubusercontent.com/63905362/187611664-3ca78729-043d-437f-aaab-eadf3614b02c.png)

**Comment by melohagan** [comment]
I understand. What you're looking for is similar to the ComboBoxes used within the Budibase builder, e.g.

![Screenshot 2022-09-01 at 16 03 44](https://user-images.githubusercontent.com/101575380/187947566-ba5d9ea9-6ec7-49ed-a0a4-41332e574c10.png)

These are currently not accessible as a component that can be added within the Builder, however with custom components being released this month, this seems like it would be reasonably easy to build!
